Sources/SwiftDocC/Infrastructure/Symbol Graph/SymbolReference.swift

```diff
@@ -69,21 +75,70 @@ public struct SymbolReference {
         path = name
     }
     
+    public init(
+        _ identifier: String,
+        unifiedSymbol: UnifiedSymbolGraph.Symbol? = nil,
+        shouldAddHash: Bool = false,
+        shouldAddKind: Bool = false
+    ) {
+        self.init(
+            identifier,
+            interfaceLanguages: unifiedSymbol?.sourceLanguages ?? [],
+            defaultSymbol: unifiedSymbol?.defaultSymbol,
+            shouldAddHash: shouldAddHash,
+            shouldAddKind: shouldAddKind
+        )
+    }
+    
+    public init(
+        _ identifier: String,
+        interfaceLanguage: SourceLanguage,
+        symbol: SymbolGraph.Symbol? = nil,
+        shouldAddHash: Bool = false,
+        shouldAddKind: Bool = false
+    ) {
+        self.init(
+            identifier,
+            interfaceLanguages: [interfaceLanguage],
+            defaultSymbol: symbol,
+            shouldAddHash: shouldAddHash,
+            shouldAddKind: shouldAddKind
+        )
+    }
+    
     /// Creates a new symbol reference with the given components and language.
     ///
     /// - Parameters:
     ///   - pathComponents: The relative path components from the module or framework to the symbol.
     ///   - interfaceLanguage: The source language of the symbol.
+    public init<SourceLanguages: Collection>(
+        pathComponents: [String],
+        interfaceLanguages: SourceLanguages
+    ) where SourceLanguages.Element == SourceLanguage {
+        self.path = pathComponents.joinedSymbolPathComponents
+        self.interfaceLanguages = Set(interfaceLanguages)
+    }
+    
     public init(pathComponents: [String], interfaceLanguage: SourceLanguage) {
         self.path = pathComponents.joinedSymbolPathComponents
-        self.interfaceLanguage = interfaceLanguage
+        self.interfaceLanguages = [interfaceLanguage]
     }
     
     /// The relative path from the module or framework to the symbol itself.
     public let path: String
     
     /// The interface language for the reference.
-    public let interfaceLanguage: SourceLanguage
+    public let interfaceLanguages: Set<SourceLanguage>
+    
+    public var interfaceLanguage: SourceLanguage {
```

Comment:
Should we mark this as deprecated and verify we're not using it this codebase anymore?